processing/0_decide_threshold.py

```diff
@@ -0,0 +1,222 @@
+    else:
+        aggregated_label.append("NA")
+
+sample_to_strain_table_full_processed["Strain type"] = aggregated_label
+
+sample_to_strain_table = sample_to_strain_table_full_processed["Strain type"].to_frame()
+
+sample_to_strain_table.head()
+# -
+
+# ## Label samples with SRA annotations
+
+# Label samples with SRA annotations
+# pao1_expression_label = pao1_expression_binned.join(
+#    sample_to_strain_table, how='left')
+pao1_expression_label = pao1_expression.merge(
+    sample_to_strain_table, left_index=True, right_index=True
+)
+pa14_expression_label = pa14_expression.merge(
+    sample_to_strain_table, left_index=True, right_index=True
+)
+print(pao1_expression_label.shape)
+pao1_expression_label.head()
+
+print(pa14_expression_label.shape)
+pa14_expression_label.head()
+
+# ## Get accessory gene expression
+
+# +
+pao1_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_annot_filename = paths.GENE_PA14_ANNOT
+
+core_acc_dict = utils.get_my_core_acc_genes(
+    pao1_annot_filename, pa14_annot_filename, pao1_expression, pa14_expression
+)
+# -
+
+pao1_acc = core_acc_dict["acc_pao1"]
+pa14_acc = core_acc_dict["acc_pa14"]
+
+# +
+# Create accessory df
+# accessory gene ids | median accessory expression | strain label
+
+# PAO1
+pao1_acc_expression = pao1_expression_label[pao1_acc]
+pao1_acc_expression["median_acc_expression"] = pao1_acc_expression.median(axis=1)
+
+# PA14
+pa14_acc_expression = pa14_expression_label[pa14_acc]
+pa14_acc_expression["median_acc_expression"] = pa14_acc_expression.median(axis=1)
+# -
+
+# Add back labels
+pao1_acc_expression["Strain type"] = pao1_expression_label["Strain type"]
+pa14_acc_expression["Strain type"] = pa14_expression_label["Strain type"]
+
+pao1_acc_expression.head()
+
+pa14_acc_expression.head()
+
+# +
+# Merge PAO1 and PA14 accessory dataframes
+pao1_pa14_acc_expression = pao1_acc_expression.merge(
+    pa14_acc_expression,
+    left_index=True,
+    right_index=True,
+    suffixes=["_pao1", "_pa14"],
+)
+
+pao1_pa14_acc_expression.head()
+# -
+
+# ## Plot distribution
+
+# +
+# Get PAO1 samples that are labeled PAO1 and non-PAO1
+pao1_sra = pao1_pa14_acc_expression.loc[
+    pao1_pa14_acc_expression["Strain type_pao1"] == "PAO1",
+    "median_acc_expression_pao1",
+]
+
+non_pao1_sra = pao1_pa14_acc_expression.loc[
+    pao1_pa14_acc_expression["Strain type_pao1"] != "PAO1",
+    "median_acc_expression_pao1",
+]
+# -
+
+pao1_threshold = 5
+
+f = sns.distplot(pao1_sra, color="grey", kde=False)
+f = sns.distplot(non_pao1_sra, color="blue", kde=False)
+plt.axvline(pao1_threshold, 0, 100, color="red")
+
+# +
+# Get PA14 samples that are labeled PA14 and non-PA14
+pa14_sra = pao1_pa14_acc_expression.loc[
+    pao1_pa14_acc_expression["Strain type_pa14"] == "PA14",
+    "median_acc_expression_pa14",
+]
+
+non_pa14_sra = pao1_pa14_acc_expression.loc[
+    pao1_pa14_acc_expression["Strain type_pa14"] != "PA14",
+    "median_acc_expression_pa14",
+]
+# -
+
+pa14_threshold = 5
```

Comment:
same question as above for the pao1_threshold definition

Reply:
Our goal here is to define a threshold to determine if a sample if PAO1 or not. We used known labels from SRA to do this. Specifically, we found a threshold that separated between the distribution of PAO1 samples (grey) vs non-PAO1 samples (blue). 

I will clarify this in the notebook.